### PR TITLE
fix(GlobalHeader): Simplify loading header

### PR DESCRIPTION
**Pull request by katiezutter**
### Overview

Fixes to header loading state.

Make the loading state consistent for desktop and mobile, to alleviate the issue of seeing two logos and a hamburger button. (Note: this is still not perfect.)

Before:

Screen Recording 2021-02-12 at 08 04 28 AM

After (see on static-sites pages at https://tayra.codecademy.com):

Screen Recording 2021-02-12 at 08 17 17 AM

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@9.3.5-alpha.4811b0.0<br/>@codecademy/styleguide@25.2.5-alpha.4811b0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60268b8f230aa1fb1d9975bf--gamut-preview.netlify.app>https://60268b8f230aa1fb1d9975bf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60268b8f230aa1fb1d9975bf>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@9.3.6-alpha.7d3b4f.0<br/>@codecademy/styleguide@25.2.6-alpha.7d3b4f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60269f8910cae5fcfed4a38e--gamut-preview.netlify.app>https://60269f8910cae5fcfed4a38e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60269f8910cae5fcfed4a38e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@9.3.6-alpha.4074fb.0<br/>@codecademy/styleguide@25.2.6-alpha.4074fb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6026a23f35682001f574af38--gamut-preview.netlify.app>https://6026a23f35682001f574af38--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6026a23f35682001f574af38>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/GlobalHeader/index.tsx [hunk 1]
```diff
@@ -95,14 +95,17 @@ const getMobileAppHeaderItems = (
...
-  const desktop = useBreakpointAtOrAbove('md');
-  return (
+  return props.type === 'loading' ? (
+    <Box height="80">
+      <AppHeader action={props.action} items={loadingHeaderItems} />
+    </Box>
+  ) : (
     <>
-      <Box display={desktop ? 'block' : 'none'} height="80">
+      <Box display={{ base: 'none', md: 'block' }} height="80">
```
Comment (rogervera) [inline packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:104]: Thanks Katie!

**Review by rogervera — APPROVED** [review]
This LGTM as far as cleaning up the two logos!
Let's keep this fix PR small in scope IMO. Any additional updates in separate PRs?

File: packages/gamut-labs/src/experimental/GlobalHeader/index.tsx [hunk 2]
```diff
@@ -95,14 +95,17 @@ const getMobileAppHeaderItems = (
 };
 
 export const GlobalHeader: React.FC<GlobalHeaderProps> = (props) => {
-  const desktop = useBreakpointAtOrAbove('md');
-  return (
+  return props.type === 'loading' ? (
+    <Box height="80">
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/experimental/GlobalHeader/index.tsx:99]: Would we not want the `display` prop to also be used for the loading state?

How about, roughly:

```tsx
<Box display=... height="80">
  {props.type === 'loading' ? ... : ...}
</Box>
```

**Review by JoshuaKGoldberg — APPROVED** [review]
+1 to using responsive CSS instead of JS effects!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@9.3.7-alpha.86980c.0<br/>@codecademy/styleguide@25.2.7-alpha.86980c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6026ca95732056242cef7408--gamut-preview.netlify.app>https://6026ca95732056242cef7408--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6026ca95732056242cef7408>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/GlobalHeader/__tests__/GlobalHeader-test.tsx [hunk 3]
```diff
@@ -91,31 +96,7 @@ const renderGlobalHeader = (props: GlobalHeaderProps) => {
...
-    .mockImplementationOnce(() => {
-      return true;
-    })
-    .mockImplementationOnce(() => {
-      return false;
-    }),
-}));
-
 describe('GlobalHeader', () => {
-  describe('responsiveness', () => {
```
Comment (katiezutter) [inline packages/gamut-labs/src/experimental/GlobalHeader/__tests__/GlobalHeader-test.tsx:106]: Checked with @saghdaey, okay to remove these.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.0.1-alpha.fbcdd8.0<br/>@codecademy/styleguide@26.0.1-alpha.fbcdd8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6026fb47f613ce00a8b31e23--gamut-preview.netlify.app>https://6026fb47f613ce00a8b31e23--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6026fb47f613ce00a8b31e23>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/GlobalHeader/__tests__/GlobalHeader-test.tsx [hunk 4]
```diff
@@ -125,8 +106,8 @@ describe('GlobalHeader', () => {
      & use 'getAllByTestId' b/c there will be duplicate elements in the DOM (since mobile & desktop render some of the same app header items) */
     test('logo', () => {
       const logoElements = screen.getAllByTestId('header-logo');
-      expect(logoElements[0]).toBeVisible(); // desktop
-      expect(logoElements[1]).not.toBeVisible(); // mobile
+      expect(logoElements[0]).not.toBeVisible(); // desktop
+      expect(logoElements[1]).toBeVisible(); // mobile
```
Comment (saghdaey) [inline packages/gamut-labs/src/experimental/GlobalHeader/__tests__/GlobalHeader-test.tsx:110]: im actually now not sure which is mobile or desktop so maybe remove those comments & we can just say there will be duplicate elements in the DOM but one will be visible one not depending on screen size

Comment (katiezutter) [reply packages/gamut-labs/src/experimental/GlobalHeader/__tests__/GlobalHeader-test.tsx:110]: For sure!

**Review by saghdaey — APPROVED** [review]
lgtm !

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.0.1-alpha.fcb286.0<br/>@codecademy/styleguide@26.0.1-alpha.fcb286.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602702f663c7de0b2aaaee28--gamut-preview.netlify.app>https://602702f663c7de0b2aaaee28--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602702f663c7de0b2aaaee28>Deploy Logs</a></p>

**Pull request merged**